### PR TITLE
scrape logs from kube-system and giantswarm namespaces only

### PR DESCRIPTION
**Pull request by marieroque**
towards https://github.com/giantswarm/roadmap/issues/2758

### CODE REVIEW

File: pkg/resource/promtail-config/promtail-config.go [hunk 1]
```diff
@@ -75,6 +76,11 @@ func GeneratePromtailConfig(lc loggedcluster.Interface) (v1.ConfigMap, error) {
 			},
 			Config: promtailConfig{
 				Snippets: promtailConfigSnippets{
+					ExtraRelabelConfigs: `# we only scrape logs from kube-system and giantswarm namespaces
```
Comment (QuentinBisson) [inline pkg/resource/promtail-config/promtail-config.go:79]: I think this should only apply to WCs though right?

Comment (marieroque) [reply pkg/resource/promtail-config/promtail-config.go:79]: Hummm I totally forgot that point

File: pkg/resource/promtail-config/promtail-config.go [hunk 2]
```diff
@@ -68,13 +70,24 @@ func ConfigMeta(lc loggedcluster.Interface) metav1.ObjectMeta {
 // the promtail extra-config
 func GeneratePromtailConfig(lc loggedcluster.Interface) (v1.ConfigMap, error) {
 
+	// Scrape logs from kube-system and giantswarm namespaces only for WC clusters
+	extraRelabelConfigs := `[]`
+	if lc.GetObservabilityBundleConfigMap() == key.WCObservabilityBundleConfigMap {
```
Comment (hervenicol) [inline pkg/resource/promtail-config/promtail-config.go:75]: As `WCObservabilityBundleConfigMap == CAPIObservabilityBundleConfigMap`, how does this make sure it does not apply to CAPI MCs?

Comment (QuentinBisson) [reply pkg/resource/promtail-config/promtail-config.go:75]: Why should it not apply to CAPI MCs though?

Comment (QuentinBisson) [reply pkg/resource/promtail-config/promtail-config.go:75]: oh nevermind

Comment (QuentinBisson) [reply pkg/resource/promtail-config/promtail-config.go:75]: We need a isWorkloadCluster condition

Comment (QuentinBisson) [reply pkg/resource/promtail-config/promtail-config.go:75]: In PMO it's mostly InstallationName == ClusterName

**Review by QuentinBisson — APPROVED** [review]
LGTM

**Review by QuentinBisson — APPROVED** [review]
LGTM

**Comment by hervenicol** [comment]
:boom: 
Both build errors look like transient errors:
- gitleak, I've had the same on another repo at the same time. Now it builds.
- Aliyun, well, that's rather usual.

We should just retry build.

**Comment by marieroque** [comment]
Tested on a `gaia` WC.

Only logs from `giantswarm` and `kube-system` namespaces:
image

**Pull request merged**